plugins/inventory/aws_ec2.py

```diff
@@ -537,14 +567,35 @@ def _get_hostname(self, instance, hostnames):
             else:
                 return to_text(hostname)
 
-    def _query(self, regions, filters, strict_permissions):
+    def _query(self, regions, include_filters, exclude_filters, strict_permissions):
         '''
             :param regions: a list of regions to query
-            :param filters: a list of boto3 filter dictionaries
+            :param include_filters: a list of boto3 filter dictionaries
+            :param exclude_filters: a list of boto3 filter dictionaries
             :param hostnames: a list of hostname destination variables in order of preference
+            :param includes_entries_matching: a list of filters, instances matching these filters are included in the result
             :param strict_permissions: a boolean determining whether to fail or ignore 403 error codes
+
         '''
-        return {'aws_ec2': self._get_instances_by_region(regions, filters, strict_permissions)}
+        instances = []
+        ids_to_ignore = []
+        for filter in exclude_filters:
+            for i in self._get_instances_by_region(
+                    regions,
+                    ansible_dict_to_boto3_filter_list(filter),
+                    strict_permissions):
+                ids_to_ignore.append(i['InstanceId'])
+        for filter in include_filters:
+            for i in self._get_instances_by_region(
+                    regions,
+                    ansible_dict_to_boto3_filter_list(filter),
+                    strict_permissions):
+                if i['InstanceId'] not in ids_to_ignore:
+                    instances.append(i)
```

Comment:
While Ansible deals well with duplicates in an inventory, it's probably better to try and filter them out.  Maybe insert into a map and then pull them back out into a list while sorting?

Reply:
Addressed by 0869bf162e9f3f37a1afb5f2a781e7ccd1a90f6a